test/framework/resource/k8s/pod/manager.go

```diff
@@ -36,6 +36,7 @@ type Manager interface {
 	DeleteAndWaitTillPodIsDeleted(context context.Context, pod *v1.Pod) error
 	GetENIDetailsFromPodAnnotation(podAnnotation map[string]string) ([]*trunk.ENIDetails, error)
 	GetPodsWithLabel(context context.Context, namespace string, labelKey string, labelValue string) ([]v1.Pod, error)
+	PatchPod(context context.Context, oldPod *v1.Pod, newPod *v1.Pod) error
```

Comment:
why not patch directly using k8sClient..since the logic is pretty simple, i don't see the value of add this wrapper :D